launcher/minecraft/mod/ResourceFolderModel.cpp

```diff
@@ -500,6 +505,67 @@ QVariant ResourceFolderModel::headerData(int section, Qt::Orientation orientatio
     return {};
 }
 
+void ResourceFolderModel::setupHeaderAction(QAction* act, int column)
+{
+    Q_ASSERT(act);
+
+    act->setText(headerData(column, Qt::Orientation::Horizontal).toString());
+}
+
+void ResourceFolderModel::saveHiddenColumn(int column, bool hidden)
+{
+    auto const setting_name = QString("UI/%1_Page/HiddenColumns").arg(id());
+    auto setting = (APPLICATION->settings()->contains(setting_name)) ?
+        APPLICATION->settings()->getSetting(setting_name) : APPLICATION->settings()->registerSetting(setting_name);
+
+    auto hiddenColumns = QVariantUtils::toList<int>(setting->get());
+    auto index = hiddenColumns.indexOf(column);
```

Comment:
I think a better saving strategy is to save the (not translated) column's header text, since saving the index will mean it will get invalid if we change ordering or add/remove columns in the future. If we use the text, then if we change a column's header text in the future, it will just stop being hidden, and I don't think we'll change one column's text to another's text, which would cause issues.

Reply:
done